module/Entra/AdditionalFunctions/Get-EntraAuditSignInLogs.ps1

```diff
@@ -0,0 +1,118 @@
+function Get-EntraAuditSignInLogs {
+    [CmdletBinding(DefaultParameterSetName = 'GetQuery')]
+    param (
+
+    [Parameter(ParameterSetName = "GetById", ValueFromPipeline = $true, ValueFromPipelineByPropertyName = $true)]
+    [System.String] $ObjectId,
+
+    [Parameter(ParameterSetName = "GetQuery", ValueFromPipeline = $true, ValueFromPipelineByPropertyName = $true)]
+    [System.Nullable`1[System.Int32]] $Top,
+
+    [Parameter(ValueFromPipeline = $true, ValueFromPipelineByPropertyName = $true)]
+    [switch] $All,
+
+    [Parameter(ParameterSetName = "GetQuery", ValueFromPipeline = $true, ValueFromPipelineByPropertyName = $true)]
+    [System.String] $Filter
+
+    )
+
+    PROCESS {
+        $customHeaders = New-EntraCustomHeaders -Command $MyInvocation.MyCommand
+        $params = @{}
+        $topCount = $null
+        $baseUri = 'https://graph.microsoft.com/v1.0/auditLogs/signIns'
+        $params["Method"] = "GET"
+        $params["Uri"] = "$baseUri"+"?"
+
+        if($null -ne $PSBoundParameters["Top"] -and  (-not $PSBoundParameters.ContainsKey("All")))
+        {
+            $topCount = $PSBoundParameters["Top"]
+            if ($topCount -gt 999) {
+                $params["Uri"] += "&`$top=999"
+            }
+            else{
+                $params["Uri"] += "&`$top=$topCount"
+            }
+        }
+
+        if($null -ne $PSBoundParameters["ObjectId"])
+        {
+            $UserId = $PSBoundParameters["ObjectId"]
+            $params["Uri"] = "$baseUri/$($UserId)"
+        }
+        if($null -ne $PSBoundParameters["Filter"])
+        {
+            $Filter = $PSBoundParameters["Filter"]
+            $f = '$' + 'Filter'
```

Comment:
What is the purpose of this line?
You are concatenating two literal strings (no expansion)?
Why not `$f = '$Filter'`?